examples/pil-fun/src/Runner.idr

```diff
@@ -118,38 +117,53 @@ run conf ctxt pp = do
+  AddFun False True "-"                $ [< Int'] ==> Just Int'
+  AddFun True  True "<"                $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True "<="               $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True "=="               $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True "||"               $ [< Bool', Bool'] ==> Just Bool'
+  AddFun True  True "&&"               $ [< Bool', Bool'] ==> Just Bool'
+  AddFun False True "!"                $ [< Bool'] ==> Just Bool'
   Enough
 
-lua5_4StdFuns : NamedCtxt
+lua5_4StdFuns : NamedCtxt Lua5_4
 lua5_4StdFuns = do
-  AddFun True  "+"  $ [< Int', Int'] ==> Just Int'
-  AddFun True  "*"  $ [< Int', Int'] ==> Just Int'
-  AddFun False "-"  $ [< Int'] ==> Just Int'
-  AddFun True  "<"  $ [< Int', Int'] ==> Just Bool'
-  AddFun True  "<=" $ [< Int', Int'] ==> Just Bool'
-  AddFun True  "==" $ [< Int', Int'] ==> Just Bool'
-  AddFun True  "or" $ [< Bool', Bool'] ==> Just Bool'
-  AddFun True  "and" $ [< Bool', Bool'] ==> Just Bool'
-  AddFun False "not"  $ [< Bool'] ==> Just Bool'
-  AddFun False "print" $ [< Int'] ==> Nothing
+  AddFun False False "print" $ [< Int'] ==> Nothing
+  AddFun True  True  "+"     $ [< Int', Int'] ==> Just Int'
+  AddFun True  True  "*"     $ [< Int', Int'] ==> Just Int'
+  AddFun False True  "-"     $ [< Int'] ==> Just Int'
+  AddFun True  True  "<"     $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True  "<="    $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True  "=="    $ [< Int', Int'] ==> Just Bool'
+  AddFun True  True  "or"    $ [< Bool', Bool'] ==> Just Bool'
+  AddFun True  True  "and"   $ [< Bool', Bool'] ==> Just Bool'
+  AddFun False True  "not"   $ [< Bool'] ==> Just Bool'
+  Enough
+
+idris2StdFuns : NamedCtxt Idris2
+idris2StdFuns = do
+  AddFun False False "printLn" ([< Int'] ==> Nothing)
+  AddFun True  True  "+"       ([< Int', Int'] ==> Just Int')    {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Int' ** Refl) True)}
+  AddFun True  True  "*"       ([< Int', Int'] ==> Just Int')    {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Int' ** Refl) True)}
+  AddFun True  True  "-"       ([< Int', Int'] ==> Just Int')    {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Int' ** Refl) True)}      
+  AddFun True  True  "<"       ([< Int', Int'] ==> Just Bool')   {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Bool' ** Refl) True)}
+  AddFun True  True  "<="      ([< Int', Int'] ==> Just Bool')   {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Bool' ** Refl) True)}
+  AddFun True  True  "=="      ([< Int', Int'] ==> Just Bool')   {lCond = Idris2Cond (IsInfix (Int' ** Int' ** Just Bool' ** Refl) True)}
+  AddFun True  True  "||"      ([< Bool', Bool'] ==> Just Bool') {lCond = Idris2Cond (IsInfix (Bool' ** Bool' ** Just Bool' ** Refl) True)}
+  AddFun True  True  "&&"      ([< Bool', Bool'] ==> Just Bool') {lCond = Idris2Cond (IsInfix (Bool' ** Bool' ** Just Bool' ** Refl) True)}
```

Comment:
Can't these `Idris2Cond ...` expressions be found with `%search`?

Reply:
Unfortunately, this can't be fixed by `%search` (since `lCond` is already auto), but the syntax was simplified by 8c6e2aa4. In the future, we can try to simplify `Idris2Cond` constructors in order to simplify proof search.

Reply:
Ok